app/reconciler/ReconcilerTypes.ts

```diff
@@ -0,0 +1,8 @@
+import type { DnsRecordType } from '@prisma/client';
+
+export interface ReconcilerCompareStructure {
```

Comment:
You don't want to mirror that same structure as AWS?  Not important that you do, but since we're creating a structure, we might consider using theirs.

Reply:
That was my original plan, but those structures are hard to compare. I'd have to do a lot of very expensive find operations, that would scale execution time esponentially O(n^2) based on the recordSet count. We would not see the issue right now, but years down the line, with many users, it sould probably cause problems.

For this structure, I'm basically using the keys as indexes, so this should scale in linear manner O(n)